examples/lib/count-star-height.js

```diff
@@ -0,0 +1,83 @@
+/* Author: Jamie Davis <davisjam@vt.edu>
+ * Description: Lib to count the star height of a regex.
+ *              This is a correct and improved version of substack's safe-regex.
+ *
+ * Star height is a heuristic for REDOS (exponential blow-up).
+ */
+
+const createRegExp = require('./create-regexp'),
+      regexpTree = require('../../');
+
+/**
+ * regex: String representation 'abc' or RegExp object /abc/
+ *        Accepts most of the Python syntax as well.
+ * options: optional -- clarify what counts as 'star height'
+ *          Keys:
+ *            countQuestionMarks -- (a+)? is not really a problem
+ *              default false (i.e. ignore ?'s)
+ *            minimumRepetitionUpperLimit -- a{0,3} is probably OK, a{0,50} is not.
+ *              default 0 (i.e. all of {x}, {x,}, {,x}, {x,y} are problematic)
+ *
+ * Returns the regex's star height: an integer >= 0.
+ */
+function countStarHeight(regex, options) {
+	/* Get an ast. */
+	const re = createRegExp(regex);
+	console.error(`re: /${re.source}/`);
```

Comment:
why is it `console.error`?

can use `console.info(...)` for normal messages

Reply:
For my use of this example I wanted stdout to only contain "the answer". Any progress updates were to stderr to make it easier for me to script around the program. I can switch to `console.info` though.